clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/KafkaInputKeyComparator.java

```diff
@@ -0,0 +1,87 @@
+package com.linkedin.venice.hadoop.input.kafka;
+
+import com.linkedin.venice.hadoop.input.kafka.avro.KafkaInputMapperKey;
+import com.linkedin.venice.serializer.FastSerializerDeserializerFactory;
+import com.linkedin.venice.serializer.RecordDeserializer;
+import java.io.IOException;
+import java.io.Serializable;
+import org.apache.avro.io.OptimizedBinaryDecoderFactory;
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.hadoop.io.DataInputBuffer;
+import org.apache.hadoop.io.RawComparator;
+import org.apache.hadoop.io.WritableComparator;
+
+
+/**
+ * This class is used to support secondary sorting for KafkaInput Repush.
+ * The key is composed by the raw key + offset, and this class will compare key part first to make it in ascending order
+ * and then compare the offset part when keys are equal to maintain an offset-descending order for the same key.
+ */
+public class KafkaInputKeyComparator implements RawComparator<BytesWritable>, Serializable {
```

Comment:
I am a bit confused here - at which phase this will be used? And since we are adopting KafkaInputKey format, why this is still on top of BytesWritable?